packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.html

```diff
@@ -1,22 +1,63 @@
+  
+      <template if:false={isCollapsed}>
+        <ul class="slds-has-dividers_around-space slds-m-vertical_xx-small">
+          <template if:true={canShowMore}>
+            <div onchange={handleKeyUp} class="facet__searchbox">
+              <lightning-input class="facet__searchbox-input slds-var-m-horizontal_small slds-p-bottom_x-small" name={labels.search} type="search" label="facet-input" variant="label-hidden" placeholder={labels.search}></lightning-input>
+            </div>
+          </template>
+          <template if:true={isAnyChecked}>
+            <button class="slds-button slds-button_icon slds-p-horizontal_small facet__collapse" value={labels.clear} onclick={clearSelections}>
+              <lightning-formatted-text class="slds-p-right_x-small" value={labels.clear}></lightning-formatted-text>
+              <lightning-icon class="slds-button__icon slds-current-color" icon-name="utility:close" size="xx-small"></lightning-icon>
+            </button>
+          </template>
 
-      <lightning-button-group>
-        <template if:true={canShowMore}>
-          <lightning-button variant="base" label={labels.showMore} title={labels.showMore} icon-name="utility:arrowdown"
-            class="slds-m-left_x-small" onclick={showMore} aria-label={showMoreFacetValuesLabel}></lightning-button>
-        </template>
-        <template if:true={canShowLess}>
-          <lightning-button variant="base" label={labels.showLess} title={labels.showLess} icon-name="utility:arrowup"
-            class="slds-m-left_x-small" onclick={showLess} aria-label={showLessFacetValuesLabel}></lightning-button>
-        </template>
-      </lightning-button-group>
+          <template if:false={isFacetSearchActive}>
+            <template for:each={values} for:item="v">
+              <c-quantic-facet-value onselectvalue={onSelect} key={v.value} item={v} is-checked={v.checked}></c-quantic-facet-value>
+            </template>
+          </template>
+
+          <template if:true={isFacetSearchActive}>
+            <template if:true={hasSearchResults}>
+              <template for:each={facetSearchResults} for:item="result">
+                <c-quantic-facet-value onselectvalue={onSelect} key={result.value} item={result} is-checked={result.checked}></c-quantic-facet-value>
+              </template>
+            </template>
+            <template if:false={hasSearchResults}>
+              <li class="slds-var-m-around_small slds-grid">
+                <span class="slds-align_absolute-center slds-p-vertical_small slds-m-bottom_medium slds-p-horizontal_xx-large facet__search-results_no-match">{noMatchesForLabel}</span>
+              </li>
+            </template>
+            <template if:true={canShowMoreSearchResults}>
+              <li class="slds-p-left_small slds-p-bottom_medium">
+                <span class="facet__search-results_more-matches">{moreMatchesForLabel}</span>
+              </li>
+            </template>
+          </template>
```

Comment:
It looks like some part here is repeated, don't really know if it's possible to simplify it?

Instead of `values` versus `hasSearchResults` maybe you can have a method:

```js
get facetValuesToDisplay() {
  return this.hasSearchResults && this.facetSearchResults || this.values || [];
}
```
And just have once:
```
<template for:each={facetValuesToDisplay} for:item="v">
  <c-quantic-facet-value onselectvalue={onSelect} key={v.value} item={v} is-checked={v.checked}></c-quantic-facet-value>
</template>
``` 
?

This is just a suggestion

Reply:
I tried to simplify that way but it turns out that it's not really better, in my opinion. I think it's better to first check `isFacetSearchActive` to distinguish between "search mode" and "normal mode" to then perform the nested checks for which buttons/messages to display in the facet depending on the number or even presence of search results.

I did however simplify the thing a bit by merging two blocks that were using the same `template if:false={isFacetSearchActive}` condition